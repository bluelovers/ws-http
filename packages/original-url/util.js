"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const url_1 = require("url");
const ip_regex_1 = require("ip-regex");
const re_v6 = ip_regex_1.v6({ exact: true });
/**
 * In case there's more than one header of a given name, we want the first one
 * as it should be the one that was added by the first proxy in the chain
 */
function getFirstHeader(req, header) {
    const value = req.headers[header];
    return (Array.isArray(value) ? value[0] : value).split(', ')[0];
}
exports.getFirstHeader = getFirstHeader;
function parsePartialURL(url) {
    const containsProtocol = url.indexOf('://') !== -1;
    const result = url_1.parse(containsProtocol ? url : 'invalid://' + url);
    if (!containsProtocol)
        result.protocol = '';
    return result;
}
exports.parsePartialURL = parsePartialURL;
function isIPv6(ip) {
    return re_v6.test(ip);
}
exports.isIPv6 = isIPv6;
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidXRpbC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbInV0aWwudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFBQSw2QkFBd0M7QUFDeEMsdUNBQThCO0FBRTlCLE1BQU0sS0FBSyxHQUFHLGFBQUUsQ0FBQyxFQUFFLEtBQUssRUFBRSxJQUFJLEVBQUUsQ0FBQyxDQUFDO0FBRWxDOzs7R0FHRztBQUNILFNBQWdCLGNBQWMsQ0FBQyxHQUU5QixFQUFFLE1BQWM7SUFFaEIsTUFBTSxLQUFLLEdBQUcsR0FBRyxDQUFDLE9BQU8sQ0FBQyxNQUFNLENBQUMsQ0FBQTtJQUNqQyxPQUFPLENBQUMsS0FBSyxDQUFDLE9BQU8sQ0FBQyxLQUFLLENBQUMsQ0FBQyxDQUFDLENBQUMsS0FBSyxDQUFDLENBQUMsQ0FBQyxDQUFDLENBQUMsQ0FBQyxLQUFLLENBQUMsQ0FBQyxLQUFLLENBQUMsSUFBSSxDQUFDLENBQUMsQ0FBQyxDQUFDLENBQUE7QUFDaEUsQ0FBQztBQU5ELHdDQU1DO0FBRUQsU0FBZ0IsZUFBZSxDQUFDLEdBQVc7SUFFMUMsTUFBTSxnQkFBZ0IsR0FBRyxHQUFHLENBQUMsT0FBTyxDQUFDLEtBQUssQ0FBQyxLQUFLLENBQUMsQ0FBQyxDQUFBO0lBQ2xELE1BQU0sTUFBTSxHQUFHLFdBQVEsQ0FBQyxnQkFBZ0IsQ0FBQyxDQUFDLENBQUMsR0FBRyxDQUFDLENBQUMsQ0FBQyxZQUFZLEdBQUcsR0FBRyxDQUFDLENBQUE7SUFDcEUsSUFBSSxDQUFDLGdCQUFnQjtRQUFFLE1BQU0sQ0FBQyxRQUFRLEdBQUcsRUFBRSxDQUFBO0lBQzNDLE9BQU8sTUFBTSxDQUFBO0FBQ2QsQ0FBQztBQU5ELDBDQU1DO0FBRUQsU0FBZ0IsTUFBTSxDQUFDLEVBQVU7SUFFaEMsT0FBTyxLQUFLLENBQUMsSUFBSSxDQUFDLEVBQUUsQ0FBQyxDQUFBO0FBQ3RCLENBQUM7QUFIRCx3QkFHQyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCB7IHBhcnNlIGFzIHBhcnNlVXJsIH0gZnJvbSBcInVybFwiO1xuaW1wb3J0IHsgdjYgfSBmcm9tICdpcC1yZWdleCc7XG5cbmNvbnN0IHJlX3Y2ID0gdjYoeyBleGFjdDogdHJ1ZSB9KTtcblxuLyoqXG4gKiBJbiBjYXNlIHRoZXJlJ3MgbW9yZSB0aGFuIG9uZSBoZWFkZXIgb2YgYSBnaXZlbiBuYW1lLCB3ZSB3YW50IHRoZSBmaXJzdCBvbmVcbiAqIGFzIGl0IHNob3VsZCBiZSB0aGUgb25lIHRoYXQgd2FzIGFkZGVkIGJ5IHRoZSBmaXJzdCBwcm94eSBpbiB0aGUgY2hhaW5cbiAqL1xuZXhwb3J0IGZ1bmN0aW9uIGdldEZpcnN0SGVhZGVyKHJlcToge1xuXHRoZWFkZXJzOiBSZWNvcmQ8YW55LCBhbnk+XG59LCBoZWFkZXI6IHN0cmluZyk6IHN0cmluZ1xue1xuXHRjb25zdCB2YWx1ZSA9IHJlcS5oZWFkZXJzW2hlYWRlcl1cblx0cmV0dXJuIChBcnJheS5pc0FycmF5KHZhbHVlKSA/IHZhbHVlWzBdIDogdmFsdWUpLnNwbGl0KCcsICcpWzBdXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBwYXJzZVBhcnRpYWxVUkwodXJsOiBzdHJpbmcpXG57XG5cdGNvbnN0IGNvbnRhaW5zUHJvdG9jb2wgPSB1cmwuaW5kZXhPZignOi8vJykgIT09IC0xXG5cdGNvbnN0IHJlc3VsdCA9IHBhcnNlVXJsKGNvbnRhaW5zUHJvdG9jb2wgPyB1cmwgOiAnaW52YWxpZDovLycgKyB1cmwpXG5cdGlmICghY29udGFpbnNQcm90b2NvbCkgcmVzdWx0LnByb3RvY29sID0gJydcblx0cmV0dXJuIHJlc3VsdFxufVxuXG5leHBvcnQgZnVuY3Rpb24gaXNJUHY2KGlwOiBzdHJpbmcpXG57XG5cdHJldHVybiByZV92Ni50ZXN0KGlwKVxufVxuIl19