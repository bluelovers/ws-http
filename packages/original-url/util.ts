import { parse as parseUrl } from "url";
import { v6 } from 'ip-regex';
import { valueFromRecord, keyFromRecord, IRecordLike } from 'value-from-record';

const re_v6 = v6({ exact: true });

/**
 * In case there's more than one header of a given name, we want the first one
 * as it should be the one that was added by the first proxy in the chain
 */
export function getFirstHeader(req: {
	headers: IRecordLike<any, any>
}, header: string): string
{
	const value = valueFromRecord<string>(header, req.headers)
	return (Array.isArray(value) ? value[0] : value).split(', ')[0]
}

export function parsePartialURL(url: string)
{
	const containsProtocol = url.indexOf('://') !== -1
	const result = parseUrl(containsProtocol ? url : 'invalid://' + url)
	if (!containsProtocol) result.protocol = ''
	return result
}

export function isIPv6(ip: string)
{
	return re_v6.test(ip)
}
