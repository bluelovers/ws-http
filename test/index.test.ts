/**
 * Created by User on 2019/6/6.
 */

import LazyURL, { ENUM_FAKE } from '../lib/index';
// @ts-ignore
import SymbolInspect from 'symbol.inspect';
// @ts-ignore
import urlParse from 'url-parse';
import { inspect } from 'util';

describe(`suite`, () =>
{
	[
		[
			['zh-TW/scripts', 'zh-TW:8080'],
			`${ENUM_FAKE.protocol}//zh-tw:8080/zh-TW/scripts`,
			`//zh-tw:8080/zh-TW/scripts`,
		],
		[
			['zh-TW/scripts', 'zh-TW'],
			`${ENUM_FAKE.protocol}//zh-tw/zh-TW/scripts`,
			`//zh-tw/zh-TW/scripts`,
		],
		[
			['zh-TW/scripts', 'http://zh-TW'],
			`http://zh-tw/zh-TW/scripts`,
			`http://zh-tw/zh-TW/scripts`,
		],
		[
			['zh-TW/scripts'],
			`${ENUM_FAKE.protocol}//${ENUM_FAKE.hostname}/zh-TW/scripts`,
			`/zh-TW/scripts`,
		],
		[
			['/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/'],
			'https://gitee.com/api/v5/repos/xxxx/xxxx/contents',
			'https://gitee.com/api/v5/repos/xxxx/xxxx/contents',
		],
		[
			['/api/v5/repos/xxxx/xxxx/contents', '/api/v5/'],
			`${ENUM_FAKE.protocol}//${ENUM_FAKE.hostname}/api/v5/repos/xxxx/xxxx/contents`,
			'/api/v5/repos/xxxx/xxxx/contents',
		],
	].forEach(test =>
	{
		// @ts-ignore
		it(inspect(test[0]), () =>
		{
			//console.log('it:inner', currentTest.title);
			//console.log('it:inner', currentTest.fullTitle());

			// @ts-ignore
			let actual = new LazyURL(...test[0] as [any, any]);
			let expected = test[1];

			expect(actual.toString()).toEqual(expected);

			//expect(actual.toString()).to.be.deep.equal(actual[Symbol.for('nodejs.util.inspect.custom')]());

			if (test[2] != null)
			{
				expect(actual.toRealString()).toEqual(test[2]);
			}

			expect(actual).toMatchSnapshot();

			let up = urlParse(actual.href);
			let up2 = urlParse(actual.toRealString());

			expect(up).toEqual(urlParse(expected as string));

			expect(up).toMatchSnapshot();
			expect(up2).toMatchSnapshot();

			//console.dir(up);
			//console.dir(up2);

			Object.keys(up)
				.forEach(name =>
				{

					if (typeof actual[name] === 'string' && !['origin'].includes(name))
					{
						// name
						expect(actual[name]).toEqual(up[name]);

						expect(actual[name]).toMatchSnapshot();
					}

				})
			;

			//console.dir(actual);
			//console.log(actual);
		});

	});

});
