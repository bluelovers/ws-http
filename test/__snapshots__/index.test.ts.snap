// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 1`] = `"fake+http://url-fake-hostname/api/v5/repos/xxxx/xxxx/contents"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 2`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "url-fake-hostname",
  "hostname": "url-fake-hostname",
  "href": "fake+http://url-fake-hostname/api/v5/repos/xxxx/xxxx/contents",
  "origin": "null",
  "password": "",
  "pathname": "/api/v5/repos/xxxx/xxxx/contents",
  "port": "",
  "protocol": "fake+http:",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 3`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "",
  "hostname": "",
  "href": "/api/v5/repos/xxxx/xxxx/contents",
  "origin": "null",
  "password": "",
  "pathname": "/api/v5/repos/xxxx/xxxx/contents",
  "port": "",
  "protocol": "",
  "query": "",
  "slashes": false,
  "username": "",
}
`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 4`] = `"fake+http:"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 5`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 6`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 7`] = `"/api/v5/repos/xxxx/xxxx/contents"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 8`] = `"url-fake-hostname"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 9`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 10`] = `"url-fake-hostname"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 11`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 12`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', '/api/v5/' ] 13`] = `"fake+http://url-fake-hostname/api/v5/repos/xxxx/xxxx/contents"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 1`] = `"https://gitee.com/api/v5/repos/xxxx/xxxx/contents"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 2`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "gitee.com",
  "hostname": "gitee.com",
  "href": "https://gitee.com/api/v5/repos/xxxx/xxxx/contents",
  "origin": "https://gitee.com",
  "password": "",
  "pathname": "/api/v5/repos/xxxx/xxxx/contents",
  "port": "",
  "protocol": "https:",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 3`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "gitee.com",
  "hostname": "gitee.com",
  "href": "https://gitee.com/api/v5/repos/xxxx/xxxx/contents",
  "origin": "https://gitee.com",
  "password": "",
  "pathname": "/api/v5/repos/xxxx/xxxx/contents",
  "port": "",
  "protocol": "https:",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 4`] = `"https:"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 5`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 6`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 7`] = `"/api/v5/repos/xxxx/xxxx/contents"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 8`] = `"gitee.com"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 9`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 10`] = `"gitee.com"`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 11`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 12`] = `""`;

exports[`suite [ '/api/v5/repos/xxxx/xxxx/contents', 'https://gitee.com/api/v5/' ] 13`] = `"https://gitee.com/api/v5/repos/xxxx/xxxx/contents"`;

exports[`suite [ 'zh-TW/scripts' ] 1`] = `"fake+http://url-fake-hostname/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts' ] 2`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "url-fake-hostname",
  "hostname": "url-fake-hostname",
  "href": "fake+http://url-fake-hostname/zh-TW/scripts",
  "origin": "null",
  "password": "",
  "pathname": "/zh-TW/scripts",
  "port": "",
  "protocol": "fake+http:",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ 'zh-TW/scripts' ] 3`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "",
  "hostname": "",
  "href": "/zh-TW/scripts",
  "origin": "null",
  "password": "",
  "pathname": "/zh-TW/scripts",
  "port": "",
  "protocol": "",
  "query": "",
  "slashes": false,
  "username": "",
}
`;

exports[`suite [ 'zh-TW/scripts' ] 4`] = `"fake+http:"`;

exports[`suite [ 'zh-TW/scripts' ] 5`] = `""`;

exports[`suite [ 'zh-TW/scripts' ] 6`] = `""`;

exports[`suite [ 'zh-TW/scripts' ] 7`] = `"/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts' ] 8`] = `"url-fake-hostname"`;

exports[`suite [ 'zh-TW/scripts' ] 9`] = `""`;

exports[`suite [ 'zh-TW/scripts' ] 10`] = `"url-fake-hostname"`;

exports[`suite [ 'zh-TW/scripts' ] 11`] = `""`;

exports[`suite [ 'zh-TW/scripts' ] 12`] = `""`;

exports[`suite [ 'zh-TW/scripts' ] 13`] = `"fake+http://url-fake-hostname/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 1`] = `"http://zh-tw/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 2`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "zh-tw",
  "hostname": "zh-tw",
  "href": "http://zh-tw/zh-TW/scripts",
  "origin": "http://zh-tw",
  "password": "",
  "pathname": "/zh-TW/scripts",
  "port": "",
  "protocol": "http:",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 3`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "zh-tw",
  "hostname": "zh-tw",
  "href": "http://zh-tw/zh-TW/scripts",
  "origin": "http://zh-tw",
  "password": "",
  "pathname": "/zh-TW/scripts",
  "port": "",
  "protocol": "http:",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 4`] = `"http:"`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 5`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 6`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 7`] = `"/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 8`] = `"zh-tw"`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 9`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 10`] = `"zh-tw"`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 11`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 12`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'http://zh-TW' ] 13`] = `"http://zh-tw/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 1`] = `"fake+http://zh-tw/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 2`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "zh-tw",
  "hostname": "zh-tw",
  "href": "fake+http://zh-tw/zh-TW/scripts",
  "origin": "null",
  "password": "",
  "pathname": "/zh-TW/scripts",
  "port": "",
  "protocol": "fake+http:",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 3`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "",
  "hostname": "",
  "href": "////zh-tw/zh-TW/scripts",
  "origin": "null",
  "password": "",
  "pathname": "//zh-tw/zh-TW/scripts",
  "port": "",
  "protocol": "",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 4`] = `"fake+http:"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 5`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 6`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 7`] = `"/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 8`] = `"zh-tw"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 9`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 10`] = `"zh-tw"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 11`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 12`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW' ] 13`] = `"fake+http://zh-tw/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 1`] = `"fake+http://zh-tw:8080/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 2`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "zh-tw:8080",
  "hostname": "zh-tw",
  "href": "fake+http://zh-tw:8080/zh-TW/scripts",
  "origin": "null",
  "password": "",
  "pathname": "/zh-TW/scripts",
  "port": "8080",
  "protocol": "fake+http:",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 3`] = `
Object {
  "auth": "",
  "hash": "",
  "host": "",
  "hostname": "",
  "href": "////zh-tw:8080/zh-TW/scripts",
  "origin": "null",
  "password": "",
  "pathname": "//zh-tw:8080/zh-TW/scripts",
  "port": "",
  "protocol": "",
  "query": "",
  "slashes": true,
  "username": "",
}
`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 4`] = `"fake+http:"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 5`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 6`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 7`] = `"/zh-TW/scripts"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 8`] = `"zh-tw:8080"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 9`] = `"8080"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 10`] = `"zh-tw"`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 11`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 12`] = `""`;

exports[`suite [ 'zh-TW/scripts', 'zh-TW:8080' ] 13`] = `"fake+http://zh-tw:8080/zh-TW/scripts"`;
